tikv-client/src/main/java/com/pingcap/tikv/predicates/ScanAnalyzer.java

```diff
@@ -323,11 +323,11 @@ public static ScanSpec extractConditions(
             continue;
           }
           if (eqMatcher.match(cond)) {
-            specBuilder.addPointPredicate(col, cond);
             if (col.isPrefixIndex()) {
               specBuilder.addResidualPredicate(cond);
               break IndexMatchingLoop;
             }
+            specBuilder.addPointPredicate(col, cond);
```

Comment:
move point predicate to here because we have added prefix index into residual predicate.

Reply:
Maybe it is index scan range's issue, simply add such condition to residual predicate will cause some cases which are able to use prefix index failed to use, further investigation is needed.